org-ql.el

```diff
@@ -825,6 +825,47 @@ A and B are Org headline elements."
             (a-priority t)
             (b-priority nil)))))
 
+;;;; Functions/Occur
+
+(defun org-ql-occur (query &optional keep-previous callback)
+  "Make a compact tree which shows all matches of the query.
+
+QUERY is an `org-ql' query sexp (quoted, since this is a
+function).
+
+The tree will show the lines where the query matches, and any other context
+defined in `org-show-context-detail', which see.
+
+When optional argument KEEP-PREVIOUS is non-nil, highlighting and
+exposing done by a previous call to `org-occur' or `org-ql-occur'
+will be kept, to allow stacking of calls to this command.
+
+Optional argument CALLBACK can be a function of no argument.  In this case,
+it is called with point at the end of the match, match data being set
+accordingly.  Current match is shown only if the return value is non-nil.
+The function must neither move point nor alter narrowing."
+  (interactive (list (read-minibuffer "Query: ")))
+  (unless keep-previous
+    (org-remove-occur-highlights nil nil t))
+  ;; (push (cons regexp callback) org-occur-parameters)
```

Comment:
From reading the docstring, I'm not sure what the purpose of `org-occur-parameters` is, but I guess we don't need it.

Reply:
The variable seem to track the history of `org-occur` calls in the buffer. Since `org-ql-occur` is not compatible with `org-occur`, you need to define another variable if you like the idea. 

IMO, the history should be expanded to all functions in `org-ql` package rather than be limited to `org-ql-occur`, but I am not sure if it should be limited to (buffer-)local or global.

Reply:
I noticed that `org-ql-agenda` has `org-ql-search-save`, which stores the configuration of `org-ql-search` to `org-ql-views` variable. It may be possible to save the configuration for `org-ql-occur` to the same variable, but you might prefer having a separate variable for the command.